src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -113,15 +113,50 @@ It is implemented recursively to ensure type stability.
     return self + row + _interactions(as, rest)
 end
 
+"""
+    trap_potential(add::SingleComponentFockAddress, pot::Array)
+
+Calculate trap potential energy of a single-component address `add`. The (precomputed) 
+potential energy per particle at each point in the lattice is given by the array `pot`.
+```math
+V = \\sum_{d=1}^D \\sum_{i=1}^M V_d x_{i,d}^2 n_i,
+```
+where ``x = (x_1, ..., x_D)`` is the distance of site `i` to the centre of the trap and 
+`n_i` is the number of particles at site `i`. `M` is the total number of sites on the 
+lattice.
+"""
+function trap_potential(add::SingleComponentFockAddress, pot::Array)
+    pe = 0.
+    for (n,i) in occupied_modes(add)
+        pe += n * pot[i]
+    end
+    return pe
+end
+
+"""
+    trap_potential(add::CompositeFS, pot::Vector)
+
+Calculate potential energy of a multicomponent address as the sum of the potential energy
+of each component. Each element of `pot` is a precomputed `Array` of potential energy for 
+that component at each lattice point.
+"""
+function trap_potential(add::CompositeFS, pot::Vector)
+    pe = 0.
+    for (i,c) in enumerate(add.components)
+        pe += trap_potential(c, pot[i])
+    end
+    return pe
+end
+
 ###
 ### HubbardRealSpace
 ###
 """
-    HubbardRealSpace(address; u=ones(C, C), t=ones(C), geometry=PeriodicBoundaries(M,))
+    HubbardRealSpace(address; u=ones(C, C), t=ones(C), v=ones(C, D), geometry=PeriodicBoundaries(M,))
```

Comment:
We should add the harmonic potential to the equation defining the Hubbard model as well.